qcodes/instrument_drivers/rohde_schwarz/ZNB.py

```diff
@@ -1,16 +1,143 @@
 import logging
+import numpy as np
 from functools import partial
 from typing import Optional, Any, Tuple
 
 from qcodes import VisaInstrument, Instrument
 from qcodes import ChannelList, InstrumentChannel
 from qcodes.utils import validators as vals
-import numpy as np
-from qcodes.instrument.parameter import MultiParameter, ArrayParameter, ParamRawDataType
+from qcodes.instrument.parameter import MultiParameter, ArrayParameter, \
+    ParamRawDataType
+from qcodes.utils.deprecate import deprecate
+
 
 log = logging.getLogger(__name__)
 
 
+class FixedFrequencyTraceIQ(MultiParameter):
+    """
+    Sweep that returns the real (I) and imaginary (Q) parts of the VNA response.
+    Requires the use of the sweep type to be set to continuous wave mode.
+    See (https://www.rohde-schwarz.com/webhelp/ZNB_ZNBT_HTML_UserManual_en
+    /ZNB_ZNBT_HTML_UserManual_en.htm) under GUI reference -> sweep softtool
+    -> sweep type tab -> CW mode
+    """
+
+    def __init__(self, name: str, instrument: "ZNBChannel",
+                 npts: int, bandwidth: int, channel: int,
+                 check_cw_sweep_first: bool = True) -> None:
+        super().__init__(name, names=("", ""), shapes=((), ()))
+        self._instrument = instrument
+        self.set_cw_sweep(npts, bandwidth)
```

Comment:
no, they do not.

Reply:
sorry, i couldn't understand - so should they call this method as well or not? :)

Reply:
I need to check the manual for the actual functioning.  But from the code it seems it's not been used for other parameters. Otherwise user code would have been giving errors already. Will check and update :)

Reply:
This function set_cw_sweep is added similar to what the set_sweep function does in the regular frequency trace. the reason for this function is that if you change the parameter npts in the VNA or the bandwidth, the setpoints for this parameter have to change accordingly. 

For the other two averaged versions of this method it is not necessary since their setpoints are always just (),(). IT therefore shouldn't be called there.

Reply:
> For the other two averaged versions of this method it is not necessary since their setpoints are always just (),()

aaa! thanks for the explanation! ( i totally misread the docstrings, my bad ).

Reply:
and this also means that the other comment suggesting an inheritance is wrong. inheritance is not the way here. but perhaps extracting the "meat" of get_raw into smth the instrument owns, could be an improvement.